src/jit/lower.cpp

```diff
@@ -228,10 +227,23 @@ GenTree* Lowering::LowerNode(GenTree* node)
                 // Vector3 return values are returned two return registers and Caller assembles them into a
                 // single xmm reg. Hence RyuJIT explicitly generates code to clears upper 4-bytes of Vector3
                 // type args in prolog and Vector3 type return value of a call
+                //
+                // RyuJIT x86 Windows: all non-param Vector3 local vars are allocated as 16 bytes. Vector3 arguments
+                // pushed as 12 bytes. For return values, a 16-byte local is allocated and the address passed
+                // as a return buffer pointer. The callee doesn't write the high 4 bytes, and we don't need to clear
+                // it either.
+                CLANG_FORMAT_COMMENT_ANCHOR;
+
+#if defined(_TARGET_64BIT_)
                 node->gtType = TYP_SIMD16;
```

Comment:
I would second this suggestion.

---
In reply to: [87865030](https://github.com/dotnet/coreclr/pull/8109#discussion_r87865030) [](ancestors = 87865030)